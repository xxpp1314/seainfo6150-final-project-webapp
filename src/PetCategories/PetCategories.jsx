import React from 'react';
import dogs from "../images/dogs.jpg";
import cats from "../images/cats.jpg";
import rabbits from "../images/rabbits.jpg";
import birds from "../images/birds.jpg";
import others from "../images/others.jpg";
import styles from "./PetCategories.module.css";
import { Link } from "react-router-dom";

const PetCategories = () => {
    return(
            <div className={styles.container}>
                <div className={styles.flex}>
                    <img className={styles.image} src={dogs} alt=""/>
                    <p className={styles.text}>   
                        <Link className={styles.text} to='/dogs'>
                            Dogs 
                        </Link>
                    </p> 
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={cats} alt=""/>
                    <p className={styles.text}>
                        <Link className={styles.text} to='/cats'>
                            Cats
                        </Link>
                    </p>
                </div>
                
                <div className={styles.flex}>
                    <img className={styles.image} src={rabbits} alt="" />
                    <p className={styles.text}> 
                        <Link className={styles.text} to='/rabbits'>
                            Rabbits
                        </Link>
                    </p>
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={birds} alt="" />
                    <p className={styles.text}> 
                        <Link className={styles.text} to='/birds'>
                            Birds
                        </Link>
                    </p>
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={others} alt=""/>  
                    <p className={styles.text}>
                        <Link className={styles.text} to='/others'>
                            Other Pets
                        </Link>
                    </p>
                </div>
        </div>
    )
}

export default PetCategories;