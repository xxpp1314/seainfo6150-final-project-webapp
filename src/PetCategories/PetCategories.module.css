.container {
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    padding: 0px 10px 50px 5px;
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
  }
  
  @media screen and (min-width: 525px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
    }
  }
  
  @media screen and (min-width: 900px) {
    .container {
      grid-template-columns: repeat(5, 1fr);
    }
  }

.flex {
    align-items: center;
    height: 50%;
    display: flex;
    flex-direction: column;  
}

.flex a:hover {
  color:rgb(160, 27, 27);
}

.image{
    padding: 10px;
    width:60%;
    height: 100%;
    border-radius: 50%;
    box-shadow:0px 0px 10px 5px grey;
    border: 5px solid maroon;
}

.text {
  padding: 20px 0px 10px 0px;
    justify-content: center;
    color:  rgb(243, 222, 222);
    text-transform: capitalize;
    text-decoration-line: none;
    font-family: 'Space Grotesk', sans-serif;
    font-weight: lighter;
}

.title {
    background-color:  rgb(243, 222, 222);
    padding: 10px 50px;
    color: maroon;
    font-weight: bold;
    text-transform: uppercase;
    font-family: 'Space Grotesk', sans-serif;
}
