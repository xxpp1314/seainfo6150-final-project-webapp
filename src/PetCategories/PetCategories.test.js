import React from "react";
import { render } from "@testing-library/react";
import PetCategories from "./PetCategories.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders the PetCategories correctly", () => {
  const { container } = render(
    <BrowserRouter>  
        <PetCategories/>
    </BrowserRouter> 
  );
  expect(container).toMatchSnapshot();
});