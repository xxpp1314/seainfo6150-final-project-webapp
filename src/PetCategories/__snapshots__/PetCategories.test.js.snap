// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PetCategories correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="dogs.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/dogs"
        >
          Dogs
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="cats.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/cats"
        >
          Cats
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="rabbits.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/rabbits"
        >
          Rabbits
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="birds.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/birds"
        >
          Birds
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="others.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/others"
        >
          Other Pets
        </a>
      </p>
    </div>
  </div>
</div>
`;
